src/DragAndDrop/Draggable/components/DraggableSource.js

```diff
@@ -62,23 +75,49 @@ const source = {
       });
     }
 
-    if (monitor.getDropResult()) {
-      const isSameGroup =
-        groupName &&
-        monitor.getItem().groupName &&
-        groupName === monitor.getDropResult().groupName;
-      const isSameContainer =
-        containerId === monitor.getDropResult().containerId;
+    const dropResult = monitor.getDropResult();
+
+    if (dropResult) {
+      const dropItem = monitor.getItem();
+      const isSameContainer = containerId === dropResult.containerId;
+      const sameGroup = isSameGroup(dropResult, groupName);
 
-      if (isSameGroup || isSameContainer) {
+      if (sameGroup || isSameContainer) {
         onDrop({
-          payload: monitor.getItem().originalItem, // original item
-          removedIndex: monitor.getItem().originalIndex, // original item index
-          addedIndex: monitor.getItem().index, // new item index
-          addedToContainerId: monitor.getDropResult().containerId, // new container for item
+          payload: dropItem.originalItem, // original item
+          removedIndex: dropItem.originalIndex, // original item index
+          addedIndex: dropItem.index, // new item index
+          addedToContainerId: dropResult.containerId, // new container for item
           removedFromContainerId: containerId, // original item container
         });
       }
+    } else {
+      const isSameContainer =
+        dragObject && containerId === dragObject.addedToContainerId;
```

Comment:
```suggestion
        containerId === dragObject.addedToContainerId;
```

`dragObject` seems to always be defined (truthy)